pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -289,7 +291,12 @@ func (l *Launcher) GetDomain(_ context.Context, _ *cmdv1.EmptyRequest) (*cmdv1.D
 	}
 
 	if len(list) > 0 {
-		if domain, err := json.Marshal(list[0]); err != nil {
+		decoratedDomain := list[0]
+		decoratedDomain.Status.OSInfo = l.domainManager.GetGuestOSInfo()
+		interfaceStatuses := getInterfaceStatuses(l, decoratedDomain)
```

Comment:
Makes sense to pass only what is needed: `l.domainManager`, `domain.Spec.Devices.Interfaces`.
I would also inline `interfaceStatuses`, but if you keep it, just remove the plural here as it is not consistent.